src/main/java/com/courtcanva/ccfranchise/services/FranchiseeService.java

```diff
@@ -0,0 +1,41 @@
+package com.courtcanva.ccfranchise.services;
+
+import com.courtcanva.ccfranchise.dtos.FranchiseeAndStaffInfoDto;
+import com.courtcanva.ccfranchise.dtos.ResponseDto;
+import com.courtcanva.ccfranchise.exceptions.ResourceNotFoundException;
+import com.courtcanva.ccfranchise.mappers.FranchiseeMapper;
+import com.courtcanva.ccfranchise.models.Franchisee;
+import com.courtcanva.ccfranchise.repositories.FranchiseeRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class FranchiseeService {
+    private final FranchiseeRepository franchiseeRepository;
+
+    private final FranchiseeMapper franchiseeMapper;
+
+    private final StaffService staffService;
+
+    @Transactional
+    public ResponseDto createFranchiseeAndStaff(FranchiseeAndStaffInfoDto franchiseeAndStaffInfoDto) {
+
+        franchiseeRepository.save(franchiseeMapper
```

Comment:
when you save, you could return a `Franchisee` and then it will contain id, then in the next line, this instance can be used to create Staff. That means you don't need to find a franchisee in staffSevice.

Reply:
thx, I refactor this logic